plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -152,17 +154,27 @@ public boolean schemaExists(ConnectorSession session, String schemaName)
 
         ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
         for (String remoteSchemaName : remoteSchemaNames) {
-            Iterable<Table> tables = bigQueryClient.listTables(DatasetId.of(projectId, remoteSchemaName), TABLE, VIEW);
-            for (Table table : tables) {
-                // filter ambiguous tables
-                boolean isAmbiguous = bigQueryClient.toRemoteTable(projectId, remoteSchemaName, table.getTableId().getTable().toLowerCase(ENGLISH), tables)
-                        .filter(RemoteDatabaseObject::isAmbiguous)
-                        .isPresent();
-                if (!isAmbiguous) {
-                    tableNames.add(new SchemaTableName(table.getTableId().getDataset(), table.getTableId().getTable()));
+            try {
+                Iterable<Table> tables = bigQueryClient.listTables(DatasetId.of(projectId, remoteSchemaName), TABLE, VIEW);
+                for (Table table : tables) {
+                    // filter ambiguous tables
+                    boolean isAmbiguous = bigQueryClient.toRemoteTable(projectId, remoteSchemaName, table.getTableId().getTable().toLowerCase(ENGLISH), tables)
+                            .filter(RemoteDatabaseObject::isAmbiguous)
+                            .isPresent();
+                    if (!isAmbiguous) {
+                        tableNames.add(new SchemaTableName(table.getTableId().getDataset(), table.getTableId().getTable()));
+                    }
+                    else {
+                        log.debug("Filtered out [%s.%s] from list of tables due to ambiguous name", remoteSchemaName, table.getTableId().getTable());
+                    }
+                }
+            }
+            catch (BigQueryException e) {
+                if (e.getCode() == 404 && e.getMessage().contains("Not found: Dataset")) {
+                    log.debug("Dataset disappeared during listing operation: %s", remoteSchemaName);
```

Comment:
BTW this seems to be a pre-existing case in JDBC metadata too. There we ignore only missing tables, not schemas.

Reply:
As far as I checked JDBC module before sending this PR, it doesn't have the same issue. Can you point out the exact issue place? The `DefaultJdbcMetadata` snippet condition is little misleading (TableNotFoundException won't be thrown), but the entire listing logic looks correct.

Reply:
It throws JDBC_ERROR wrapping the original exception that was encountered instead of ignoring failures during listing.

See BaseJdbcClient#getTableNames - https://github.com/trinodb/trino/blob/338425db5019102630bcfe5e6894e2677b05ebae/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L226-L228

Reply:
The `getTables()` in the try block calls `DatabaseMetaData.getTables()` method, so it won't throw an exception even if the schema doesn't exist in my understanding. `IdentifierMapping` might lead to the failure though.

Reply:
Sorry if it was unclear. I meant that the existing JDBC code can fail if `DatabaseMetaData#getTAbles` fails which we know to be the case with MemSQL for example. It's purely theoretical but it can happen IIUC?